### PR TITLE
ncm-ofed: new options and cleanup

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

**Comment by stdweird** [comment]
jenksin test this please

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 1]
```diff
@@ -7,109 +7,119 @@ declaration template components/ofed/schema;
...
+@documentation{openib options}
 type component_ofed_openib_options = {
     "onboot" : boolean = true
 
-    ## MAD datagrams priority
+    @{MAD datagrams priority}
     "renice_ib_mad" : boolean = false
 
-    ## disable for large clusters
+    @{disable for large clusters}
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: Disable what? Why?

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 2]
```diff
@@ -7,109 +7,119 @@ declaration template components/ofed/schema;
...
     "set_ipoib_cm" : boolean = true
     "set_ipoib_channels" : boolean = false
+    @{IPoIB MTU setting for CM}
+    "ipoib_mtu" : long = 32 * 1024
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: Validation? Must be positive?

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 3]
```diff
@@ -7,109 +7,119 @@ declaration template components/ofed/schema;
...
-    ## SRP High Availability
+    @{SRP High Availability}
     "srpha_enable" : boolean = false
     "srp_daemon_enable" : boolean = false
 
-    ## IPoIB MTU setting
-    "ipoib_mtu" : long = 32*1024
-
-    # autotuning
+    @{autotuning}
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: ?

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 4]
```diff
@@ -7,109 +7,119 @@ declaration template components/ofed/schema;
...
 
-## openib modules (OPENIBMODULES)
+@{openib modules to load}
 type component_ofed_openib_modules = {
     "ucm" : boolean = false
     "umad" : boolean = true
     "uverbs" : boolean = true
 
-    ## RDAM CM (connected mode and unreliable datagram)
+    @{RDMA modes (connected mode and unreliable datagram)}
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: Separate annotations for each

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 5]
```diff
@@ -1,115 +1,121 @@
...
+@documentation{openib hardware modules to load}
 type component_ofed_openib_hardware = {
-    ## Mellanox
-    ## Inifinihost III
+    @{Mellanox Inifinihost III}
     "mthca" : boolean = false
-    ## Connectx et al
+    @{Mellanox ConnectX}
     "mlx4" : boolean = false
     "mlx5" : boolean = false
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: The annotation above doesn't get associated with this option, perhaps:

```pan
    @{Mellanox ConnectX 4}
    "mlx4" : boolean = false
    @{Mellanox ConnectX 5}
    "mlx5" : boolean = false
```

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 6]
```diff
@@ -1,115 +1,121 @@
...
 
-    ## Mellanox ethernet
+    @{Mellanox ethernet-only}
     "mlx_en" : boolean = false
 
-    ## Qlogic
+    @{Qlogic}
     "ipath" : boolean = false
     "qib" : boolean = false
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: Same here, but I note that `ipath` is deprecated and has been removed (see https://github.com/torvalds/linux/commit/b85d9905a7ca128f24e3a4e60ff2a1b0cd58ae7c)

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 7]
```diff
@@ -1,115 +1,121 @@
...
+    @{Reliable datagram socket}
     "rds" : boolean = false
 
-    ## ISCSI RDMA
+    @{ISCSI RDMA}
     "iser" : boolean = false
 
     "mlx4_vnic" : boolean = false
     "mlx4_fc" : boolean = false
+    "mlx4_en" : boolean = false
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: `@{Mellanox ConnectX Ethernet}` ?

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 8]
```diff
@@ -1,115 +1,121 @@
...
-    ## Reliable datagram socket
+    @{Reliable datagram socket}
     "rds" : boolean = false
 
-    ## ISCSI RDMA
+    @{ISCSI RDMA}
     "iser" : boolean = false
 
     "mlx4_vnic" : boolean = false
     "mlx4_fc" : boolean = false
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: `@{Mellanox ConnectX FibreChannel over Ethernet}` ?

File: ncm-ofed/src/main/pan/components/ofed/schema.pan [hunk 9]
```diff
@@ -1,115 +1,121 @@
...
 
-    ## Reliable datagram socket
+    @{Reliable datagram socket}
     "rds" : boolean = false
 
-    ## ISCSI RDMA
+    @{ISCSI RDMA}
     "iser" : boolean = false
 
     "mlx4_vnic" : boolean = false
```
Comment (jrha) [inline ncm-ofed/src/main/pan/components/ofed/schema.pan]: `@{Mellanox ConnectX Ethernet over Infiniband}` ?

**Comment by jrha** [comment]
@stdweird *poke*

**Comment by stdweird** [comment]
@jrha done

**Comment by jrha** [comment]
Awesome, thanks.

**Pull request merged**